lib/db/cms/services.ts

```diff
@@ -99,10 +99,36 @@ const deleteOfferingSection = async (
     }
 }
 
+const addOfferingEvent = async (
+    id: string,
+    section: string,
+    data: { description: string; title: string; img: string }
+): Promise<ServicesProps> => {
+    try {
+        const services = await getServices()
+        const updatedOfferings = [...services.offerings].map(offering => {
```

Comment:
Technically, you can always do this. The reason to spread is it _should_ save the original data in case there were something to go wrong within the map or overwrite the original data.